File: public/aramex/RatesCalculatorAPI/aramex-rates-calculator-api-sample-code/RateCalculatorClientCSharp/RateCalculatorClientCSharp/TestCase.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using RateCalculatorClient.RateCalculatorReference ;

namespace RateCalculatorClient
{
    public class TestCase
    {
        #region "Members"
            private int _ID = 0;
            private string _Text = string.Empty;
            private string _Description = string.Empty;
            private RateCalculatorRequest _RateRequest = null;
            private RateCalculatorResponse _RateResponse = null;
        #endregion

        #region "Properties"
            public int ID
            {
                get { return _ID; }
                set { _ID = value; }
            }

            public string Text
            {
                get { return _Text; }
                set { _Text = value; }
            }

            public string Description
            {
                get { return _Description; }
                set { _Description = value; }
            }

            public RateCalculatorRequest RateRequest
            {
                get { return _RateRequest; }
                set { _RateRequest = value; }
            }

            public RateCalculatorResponse RateResponse
            {
                get { return _RateResponse; }
                set { _RateResponse = value; }
            }
        #endregion

        #region "Constructors"
            public TestCase(int ID, 
                            string Text, 
                            string Description,
                            RateCalculatorRequest RateRequest,
                            RateCalculatorResponse RateResponse)
            {
                _ID = ID;
                _Text = Text;
                _Description = Description;
                _RateRequest = RateRequest;
                _RateResponse = RateResponse;
            }
        #endregion
    }
}

File: public/aramex/RatesCalculatorAPI/aramex-rates-calculator-api-sample-code/RateCalculatorClient/RateCalculatorClient/TestCase.vb
Imports RateCalculatorClient.RateCalcService

Public Class TestCase

#Region "Members"
    Private _ID As Integer = 0
    Private _Text As String = String.Empty
    Private _Description As String = String.Empty
    Private _RateRequest As RateCalculatorRequest = Nothing
    Private _RateResponse As RateCalculatorResponse = Nothing
#End Region

#Region "Properties"
    Public Property ID As Integer
        Get
            Return _ID
        End Get
        Set(ByVal value As Integer)
            _ID = value
        End Set
    End Property

    Public Property Text As String
        Get
            Return _Text
        End Get
        Set(ByVal value As String)
            _Text = value
        End Set
    End Property

    Public Property Description As String
        Get
            Return _Description
        End Get
        Set(ByVal value As String)
            _Description = value
        End Set
    End Property

    Public Property RateRequest As RateCalculatorRequest
        Get
            Return _RateRequest
        End Get
        Set(ByVal value As RateCalculatorRequest)
            _RateRequest = value
        End Set
    End Property

    Public Property RateResponse As RateCalculatorResponse
        Get
            Return _RateResponse
        End Get
        Set(ByVal value As RateCalculatorResponse)
            _RateResponse = value
        End Set
    End Property
#End Region

#Region "Constructors"
    Public Sub New(ByVal ID As Integer, _
                   ByVal Text As String, _
                   ByVal Description As String, _
                   ByVal RateRequest As RateCalculatorRequest, _
                   ByVal RateResponse As RateCalculatorResponse)
        _ID = ID
        _Text = Text
        _Description = Description
        _RateRequest = RateRequest
        _RateResponse = RateResponse
    End Sub
#End Region

End Class
